Sync v2-9-stable with v2-9-test to release 2.9.3

utkarsharma2 opened the pull request:
Time for 2.9.3rc1!

jedcunningham committed:
Fixup python client release guide (#39182)

This also fixes 1 line in the vote email for Airflow core.

(cherry picked from commit 665da50567413485c46dca3f9fd6f6ba7e288e97)

dstandish committed:
Rename "try_number" increments that are unrelated to the airflow concept (#39317)

This avoids some confusion and false positives when looking for try_number mutation in the database.

(cherry picked from commit 4fa9fe119e5aae639c942668f40a15a6ef0f7f67)

shahar1 committed:
Introduce anonymous credentials in GCP base hook (#39695)

(cherry picked from commit 27b3a22e341468855c4ef368015ad946a59aa2e3)

eladkal committed:
Pin requests due to incompatibility with docker-py (#39740)

* Pin requests due to incompatibility with docker-py

* fix

(cherry picked from commit 1da7f1f4339241c0a0aa59e158c4cd480c1b2ddf)

potiuk committed:
Limit requests in botocore upgrade test (#39747)

The botocore upgrade test brings also requests to latest version
even if it is limited in `[docker]` extra after #39740. This PR
adds similar limit when botocore is upgraded.

(cherry picked from commit 39269d6a71f9696cbe0053e23fb965a59ca555c7)

dirrao committed:
uv version bump to 0.2.3 (#39848)

* uv version bump 0.2.3

* uv version bump 0.2.3

(cherry picked from commit cda4a8604fc755d89d1e9fbfaa1df7c40eb03184)

eladkal committed:
Bump docker version to 7.1.0 (#39839)

* Bump docker and requests versions

* fixup! Bump docker and requests versions

---------

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>
(cherry picked from commit aec3f802783c42f8544c631d08e3b226d41cdc39)

techolga committed:
Improve trigger UI for string array format validation (#39993)

* Improve trigger UI for string array format validation

The schema argument in a Param definition is used to detemine the
relevant UI element. When the type of a Param is `array` a simple
textarea is rendered. When additionally a schema for the items of
the array is provided, the UI element previously switched to a
CodeMirror box that expects pure JSON inputs. For strings a textarea
is presented now, to keep the same behaviour than without a format
definition.

Besides that the jinja template now accesses the `items` member of the
schema correctly to avoid problems with empty `items` arrays.

* Add details about the attribute `items` for the type `array`

(cherry picked from commit 914bccc387233b799309499c727074d15e0afbd4)

bangjiehan committed:
Update index.rst (#40040)

Remove the word "docker-compose" in PyPI section.

(cherry picked from commit 32cf8cf37d57b2d414a8c1dc2ac73449a4cbacd8)

bangjiehan committed:
Fix typos in overview.rst (#40061)

(cherry picked from commit 9bcf665570f3b87eeb5932500c1ae10391ef067d)

uzhastik committed:
fix typo and path to licences folder (#40067)

(cherry picked from commit fcb9f87c55a30da2c727881ff6cf47aca25dae78)

Update commands hashs

jedcunningham committed:
Docs: Simpler command to check local scheduler is alive (#40074)

(cherry picked from commit 16e0182025c27cc26c5fd76fb451de3073f91a49)

bangjiehan committed:
Fix grammar in dags.rst (#40085)

* Fix grammer dags.rst

To say "add Operator/Task to the DAG" even though DAG is an argument of Operator constructor, and use the word "with statement" explicitly instead of context manger to be friendly to Python beginner.

* Fix grammer in dags.rst

Because there is a direction in relationship, it's somehow confuse to say that they depend on each other.

* Fix grammer in dags.rst

(cherry picked from commit b03aa6caa076171337054c3adafe74e775d92453)

noel-gallagher committed:
fix typo in example_params_ui_tutorial (#40094)

(cherry picked from commit a31b10edda0fa07f5c8e75c40842c909e8856b64)

bangjiehan committed:
Remove apt-transport-https because It is a dummy transitional package(#40015)

Remove apt-transport-https because It is a dummy transitional package.

(cherry picked from commit 0e8d823a9f5174bb9f11ee6d69d71c3e0c0a2e75)

potiuk committed:
Fix credentials intialization revealed by mypy version of google auth (#40108)

(cherry picked from commit 013523e5d32dd4169f0535804e94762bd4463f79)

jedcunningham committed:
Use a join for TI notes in TI batch API endpoint (#40028)

(cherry picked from commit 4723dbe99a8240e503ad7bd7efb6da916b7addfc)

potiuk committed:
Upgrade to latest installers (#40107)

(cherry picked from commit 0c51bd6ab6ef265fafea898fdb1ec62ce76e5032)

jedcunningham committed:
Fix param order in PythonOperator docstring (#40122)

(cherry picked from commit b49db97b8e2fd5fbeaef469cec2a1343e21a411b)

potiuk committed:
Fix openai 1.32 breaking openai tests (#40110)

The new openai release adds new required parameter `parallel_tool_calls`
and our mock did not have it. Bumping version and adding the parameter
should solve the problem. It turned out also that uv==0.2.6 introduced
a bug for lowest-direct resolution, so we need to limit it to 0.2.5.

Details in https://github.com/astral-sh/uv/issues/4136

(cherry picked from commit 65dbf86f72ed7be779e7dadd8e8e57c1216c7c07)

jedcunningham committed:
Add warning to deprecated api docs that access control isn't applied (#40129)

(cherry picked from commit 85be1867a690726b4556a58656597a0606ba09dc)

potiuk committed:
Upgrade to codecov@v4 (#40128)

Codecov3 uses deprecated node version and generates warnings.

(cherry picked from commit 27579cb77354957618ae0b65f76bf32d49e9826b)

bangjiehan committed:
Fix highlight of example code in dags.rst (#40114)

highlight task_group decorator and dependency between task and task group.

(cherry picked from commit 340d6b0dde6abcbfb7e440052bc9e9d771d014da)

kacpermuda committed:
Fix typo in providers_manager.py (#40157)

Signed-off-by: Kacper Muda <mudakacper@gmail.com>
(cherry picked from commit 072c63b5397b1333bdb6fd8ed1bf58f49366e8d1)

bangjiehan committed:
Fix minor typo in dags.rst (#40169)

(cherry picked from commit e9e3c937f31f07b9235ffc74ac70482c864800e9)

potiuk committed:
Upgrade build installers and dependencies (#40177)

We can now remove the 0.2.5 limit on uv after the issue we opened
https://github.com/astral-sh/uv/issues/4136 has been addressed in 0.2.10

(cherry picked from commit 7c4ea23b4d5aa4fd757c0c3caef358d4cab486e9)

dependabot[bot] committed:
Bump braces from 3.0.2 to 3.0.3 in /airflow/www (#40180)

Bumps [braces](https://github.com/micromatch/braces) from 3.0.2 to 3.0.3.
- [Changelog](https://github.com/micromatch/braces/blob/master/CHANGELOG.md)
- [Commits](https://github.com/micromatch/braces/compare/3.0.2...3.0.3)

---
updated-dependencies:
- dependency-name: braces
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
(cherry picked from commit 1533ba6bf32b963dda92edc0f6363a2f49cd5027)

Usiel committed:
Ensures DAG params order regardless of backend (#40156)

* Ensures DAG params order regardless of backend

Fixes https://github.com/apache/airflow/issues/40154

This change adds an extra attribute to the serialized DAG param objects which helps us decide
the order of the deserialized params dictionary later even if the backend messes with us.

I decided not to limit this just to MySQL since the operation is inexpensive and may turn
out to be helpful.

I made sure the new test fails with the old implementation + MySQL. I assume this test will be
executed with MySQL somewhere in the build actions?

* Removes GitHub reference

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

* Serialize DAG params as array of tuples to ensure ordering

Alternative to previous approach: We serialize the DAG params dict as a list of tuples which _should_ keep their ordering regardless of backend.

Backwards compatibility is ensured because if `encoded_params` is a `dict` (not the expected `list`) then `dict(encoded_params)` still works.

* Make backwards compatibility more explicit

Based on suggestions by @uranusjr with an additional fix to make mypy happy.

---------

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit 2149b4dbee8fb524bb235280aaef158afaec8d4a)

raphaelauv committed:
doc: metrics allow_list complet example (#40120)

Co-authored-by: raphaelauv <raphaelauv@users.noreply.github.com>
(cherry picked from commit 205ad57d040201c5b0690d100719207695028fc2)

phi-friday committed:
Fix import future annotations in venv jinja template (#40208)

(cherry picked from commit d5a75446a62ba8804879177ce394c7456adaa4d3)

bangjiehan committed:
Fix typo in tasks.rst (#40228)

(cherry picked from commit 25872951913399236b35398226a3111e10b8b2cd)

htpawel committed:
Fix dag task scheduled and queued duration metrics (#37936)

(cherry picked from commit bffb7b08c300a79ace933b396a91220af5afa919)

merobi-hub committed:
Add slightly more detailed guidance about upgrading to the docs (#40227)

---------

Signed-off-by: merobi-hub <merobi@gmail.com>
(cherry picked from commit 16b17f72531d87cd51f2b27ce9958a9b6be52be1)

bangjiehan committed:
Fix typo in index.rst (#40245)

(cherry picked from commit f5079dbb043579fc3f5ebf14af72ba1a12764d2f)

potiuk committed:
Update to uv 0.2.11 (#40244)

(cherry picked from commit 3014165506505a3388d2e00277148d828ddb5ab3)

softyoungha committed:
Change `httpx` to `requests` in `file_task_handler` (#39799)

* Change httpx to requests in file_task_handler

- httpx does not support CIDRs in NO_PROXY
- simply, convert httpx to requests, issues done
- related issue: https://github.com/apache/airflow/issues/39794

* Add cidr no_proxy test test_log_handlers.py

* Apply monkeypatch fixture

---------

Co-authored-by: scott-py <scott.py@kakaocorp.com>
(cherry picked from commit 1ddadf59b8089f71d3c0f153aa62112d073039fc)

potiuk committed:
Fix typing in telegram provider (#40255)

The python-telegram-bot new version has typing added and we should
pass the right dict type to it.

(cherry picked from commit bc4ca9d87975065497e55fcd21a31d784f8fdf05)

potiuk committed:
Working fix for typing in telegram provider (#40258)

The python-telegram-bot new version has typing added and we should
pass the right dict type to it. The #40255 was an unsuccessful
attempt to fix it, this one actually fixes it.

(cherry picked from commit 1451bacc2e696167224621243600e57780650d71)

potiuk committed:
Fix typo when uninstalling weaviate for Pydantic 1 tests (#40259)

Weaviate 4 requires Pydantic v2, and we currently limit weaviate to
4+. There was already removal of weaviate for Pydantic 1, but it
had a typo which caused the weaviate-client not to be uninstalled.

(cherry picked from commit cb372e5c8a81e8b873c504e3744492cda7034111)

bangjiehan committed:
Fix typo in xcoms.rst (#40265)

(cherry picked from commit 56904394b9162644bf4b2a79651ee7d6b9fc05b6)

bbovenzi committed:
Swap dag run link from legacy graph to grid with graph tab (#40241)

(cherry picked from commit 7ecbf39878b65deb4a5af108171635b0cd83d478)

jedcunningham committed:
Fix compatibility with old MySQL 8.0 (#40314)

(cherry picked from commit 5221196df4d5bd9e534db34ca683cf85f2e20fcf)

Update diagram

potiuk committed:
Fix mypy problems in new waeviate client (#40330)

The new waeviate client (4.6.5) has more typing information and it
causes Waeviate Hook to fail mypy check. This PR fixes it.

(cherry picked from commit d048122eb1899a5d5d16ad67b876d7e7c6982e66)

potiuk committed:
Update uv to 0.2.13 (#40344)

(cherry picked from commit 8217bec8acfa24e0a7528735d5ede30e3da075d0)

ephraimbuddy committed:
Add `[webserver]update_fab_perms` to deprecated configs (#40317)

* Add `[webserver]update_fab_perms` to deprecated configs

`[webserver]update_fab_perms` is deprecated in favour of `[fab]update_fab_perms` and has been a breaking change since 2.9.0. This PR adds the config to the deprecated config list to properly inform users and have both options work at the moment

* Move provider config loading into init_appbuilder

* Remove deprecated webserver config options from config.yml

* Remove fallbacks

* Remove update_fab_perms and return the provider initialization

* Include FAB config in core configuration reference

* Update docs/apache-airflow/faq.rst

* Fix doc build error

(cherry picked from commit e24b7c1de319a4032e5c682a3f80e38b0dec9248)

potiuk committed:
Upgrade `pip` to just-released 24.1 (#40367)

Beta versions have been tested previously with #40184

(cherry picked from commit 9dc521e66dab07041925b3d12a9f65e629eaf7f8)

fredthomsen committed:
Fix TriggeredDagRunOperator triggered link (#40336)

* Fix TriggeredDagRunOperator triggered link

Link to the specific dag run id that was triggered, as opposed to the
logical date, such that the instance that was triggered on the DAG grid
UI.

* Add comment regarding triggered dag xcom

(cherry picked from commit 806bb80ce43dcd79ab57b040151a06ff679f2cf1)

potiuk committed:
Upgrade to latest version of hatchling as build dependency (#40387)

(cherry picked from commit 0779d9f74f9627d72cdefb123cc5f68d061f7460)

le-chartreux committed:
fix typos in example of DAG parameters (#40400)

* fix: typo in example_params_ui_tutorial.py (des -> does)

Modify "des not allow" to "does not allow"

* fix typo in example of DAG parameters

Modify a line of example_params_ui_tutorial.py to remove a trailing '.e' at the end of a comment.
Before:
# Note: Value display mapping does not need to be complete.s
After:
# Note: Value display mapping does not need to be complete

(cherry picked from commit e8a8208477a4dd91d4b89b3ca5c964a3ea69d2ab)

marekhanus committed:
docs: fix typo in upgrading.rst (#40399)

(cherry picked from commit 3133c33684d6fc97842e697c65746da11a690c08)

potiuk committed:
Update uv to 0.2.15 (#40411)

(cherry picked from commit bd18585a851bc048fa403108614de5bd47154cac)

DLT1412 committed:
fix(statsd): handle unsupport operand int + str when value of tag is int (job_id) (#40407)

(cherry picked from commit 19baf9c60553ca7ae5af466f94f2bb23ee306df8)

ferruzzi committed:
Remove "experimental" banner for OTel Metrics (#40286)

(cherry picked from commit d5c1fc28309ab53c12bcc6d0f95d31b0361e25a5)

aritra24 committed:
extra being passed to SQLalchemy (#40391)

The edit prefills an empty extra field (None)
with a empty dictionary, which is decoded
to an extra field being passed in the con url.

(cherry picked from commit a856ed4fb04352798664e79b9ff32b9f375b1d75)

potiuk committed:
Upgrade to latest `pip` and `uv` version (#40463)

(cherry picked from commit a956e30bce708c6e422953dc764f2ad030dc5403)

amoghrajesh committed:
Validating provider description for urls in provider list view (#40475)

* Validating provider description for urls in provider list view

* adding unit tests

---------

Co-authored-by: adesai <adesai@cloudera.com>
(cherry picked from commit f18f48492dc69f392e45567580b6ddb0c070ea58)

dirrao committed:
uv version bump 0.2.18 (#40501)

(cherry picked from commit da5fed9c5dcb575485536103f7dda45878b8425a)

potiuk committed:
Fix latest botocore tests to not conflict with apache-beam provider (#40502)

The apache-beam provider has certain expectations about requests
that need to be included when we manually bump botocore to the
latest version - to test if the latest version still works with
non-aiobotocore tests.

Added the requests reqiurement from apache.beam and also improved
the comment on botocore update to explain where the requirements
come from - also added displaying the actual command that is
run to upgrade botocore..

(cherry picked from commit 54dfead4ccfe7a5b41eab3433c5a38159434e014)

RNHTTR committed:
revamp some confusing log messages (#40334)

(cherry picked from commit 34e7cab949dcabdad04d201d05de2cd1a6a24b29)

potiuk committed:
Add more precise description of maskiing sensitive field names (#40512)

(cherry picked from commit dc038892d0c8313e144ae2a1d08a3ab7d970e810)

boushphong committed:
Fix dag (un)pausing won't work on environment where dag files are missing (#40345)

closes: #38834

#38265 added bulk pause and resume of DAGs. However, this PR seems to reuse the cli util method that collects DAGs from the default dag folder but not from the metadata DB. Hence, this would cause the unpause command to fail on environments where the dag folder is missing.
(cherry picked from commit e3d62c3a59358d7d2d1bc19f32a3b13c61998c4d)

potiuk committed:
Add notes about passing secrets via environment variables (#40519)

(cherry picked from commit 07e6eb8e36d2a1c4a7f2e408821997a06d57d0fe)

bbovenzi committed:
Fix calendar view scroll (#40458)

(cherry picked from commit acdac24e3f40e28ff1ac067f9038364bcc96cae8)

potiuk committed:
Update to latest pre-commit (#40529)

(cherry picked from commit de60b7bd499d4a87c5f4c076bcb3ec7385b49874)

Lee-W committed:
fix: disable jinja2 rednering for doc_md (#40522)

(cherry picked from commit 8159f6e24704f5e0e3b3217cf79ecf5083dce531)

potiuk committed:
Update trove classifiers to the latest version as build dependency (#40542)

(cherry picked from commit fb4bcaabf9b0a1dd8ce883bc6cbec5d9eebb9d4d)

utkarsharma2 committed:
Update version to 2.9.3

Dev-iL committed:
Add a note and an example clarifying the usage of DAG-level params (#40541)

(cherry picked from commit 8bdd0e72ffef0461b4d0828de7fe49def2515804)

utkarsharma2 committed:
Update RELEASE_NOTES.rst

Update release notes

Update RELEASE_NOTES.rst

Update release notes

Update RELEASE_NOTES

Fix static checks

Update RELEASE_NOTES.rst

Updated release date

Fix static check

update release notes

smlee729 committed:
Recognize quotes when parsing urls in logs (#40508)

---------

Co-authored-by: Jacob Lee <seungmin_lee3@apple.com>
(cherry picked from commit 2f9b23d568fb12edcb48fefd010fef8ec45dd12e)

eladkal committed:
Further clarify triage process (#40536)

* Further clarify triage process

* update

* update

(cherry picked from commit 90a6d7f9ead21bebc750baf38817a8f1600b3d17)

simi committed:
Update best-practices.rst (#40505)

- minor rephrase of example

(cherry picked from commit 5b2becbfb45d64d527295a54fc8830c6d4374616)

ayushjain2809 committed:
Remove logging for "blocked" events (#40446)

(cherry picked from commit ed6849eb604ffbb1f5a0efcbbbe612440e04902e)

potiuk committed:
Upgrade to another version of trove-classifier (new CUDA classifiers) (#40564)

(cherry picked from commit 888b4a0b18c2a2e94a5d62a69f72835a463275ca)

vincbeck committed:
Fix bug in `SchedulerJobRunner._process_executor_events` (#40563)

(cherry picked from commit a8c4830d9728e9e25d1164e124004eb42cda6e58)

khalidzohaib committed:
MSSQL-DOC-FIX (#40565)

Co-authored-by: khalidzohaib <hello@zohaibkhalid.com>
(cherry picked from commit 4d9746173988020cf9e62c48921e787881fe3e91)

steffidlf committed:
Fix import statement for DatasetOrTimetable example (#40601)

Fix import statement for DatasetOrTimetable example

(cherry picked from commit a984d973aa94df9c39e423bc65da382089bd5319)

CaselIT committed:
Update serializers.rst (#40597)

Mention that ``serialize`` cannot return bytes

(cherry picked from commit 49ded9f5330a6edaf88c1572098d94c643ccee77)

amoghrajesh committed:
Updating airflow download links to CDN based links (#40618)

(cherry picked from commit 77a9850556c3035d1798d4497705780c0a94774c)

abhishekbhakat committed:
Skip checking subdagslist if taskinstance state is skipped (#40578)

* Skip checking subdagslist if taskinstance state is skipped

* linter fixes

(cherry picked from commit ec5da924402709738101508b1ece15a90fd76e27)

dirrao committed:
uv version bump 0.2.22 (#40641)

(cherry picked from commit 296187ed34d7565b807c17f64432ff74981e0ee3)

potiuk committed:
Update `pip` to just upgraded 24.1.2 (#40647)

(cherry picked from commit 9a8b2d2eeb73b83eb93baf4257ac457954e906be)

utkarsharma2 committed:
Fix typo in condition that caused provider tests to be run v2 branch

mimic: https://github.com/apache/airflow/pull/40654/

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,68 @@
+Support for OpenTelemetry Metrics is no longer "Experimental" (#40286)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Official support for OpenTelemetry
```
I dont think this adds more information than the row above?
Maybe we should say something like "experimintal support for OpenTelemetry was added in ____ since then fixes and improvments were added and now we announce the feature as stable"

utkarsharma2 replied:
Thanks @eladkal I have updated it in the RELEASE_NOTES.rst.

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,68 @@
+- Change ``httpx`` to ``requests`` in ``file_task_handler`` (#39799)
+- Fix import future annotations in venv jinja template (#40208)
+- Ensures DAG params order regardless of backend (#40156)
+- Fix highlight of example code in dags.rst (#40114)
```
This is doc only change

utkarsharma2 replied:
Updated in the RELEASE_NOTES.rst

ephraimbuddy commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,68 @@
 
 .. towncrier release notes start
 
+Airflow 2.9.3 (2024-07-02)
```
```suggestion
Airflow 2.9.3 (2024-07-12)
```

utkarsharma2 replied:
Updated in the RELEASE_NOTES.rst

a1ea321 committed:
Add warning about the PostgresOperator being deprecated (#40662)

PostgresOperator's reference page says it's deprecated. I am not knowledgeable enough to write a tutorial. I think if someone reads the guide (I linked to in my change) they will be able to use this tutorial as well.

(cherry picked from commit a10ac28755475f445a61e9e15f092dcd205ed1f2)

potiuk committed:
Fix typing issue with new version of types-docutils released` (#40727)

(cherry picked from commit 97b88fd825007d61fd245b065cd72c8de8c49447)
